framework/util/platform.h

```diff
@@ -614,7 +628,14 @@ inline bool StringContains(const char* text, const char* substring)
 
 inline int32_t FilePuts(const char* char_string, FILE* stream)
 {
-    return fputs(char_string, stream);
+    int result = 0;
+    int err    = 0;
+    do
+    {
+        result = fputs(char_string, stream);
+        err    = ferror(stream);
+    } while (result == EOF && (err == EWOULDBLOCK || err == EINTR || err == EAGAIN));
```

Comment:
Should this be result != EOF?

Reply:
I have little experience with fputs, but it seems correct. The man page seems to indicate that any condition where it does not write the full desired character will make it return EOF, which would be our case here. So EOF does not mean literal "end of file" (not sure what that would mean for a write operation anyways).

However, there is a more subtle error in my patch, as it does not handle incomplete reads or writes correctly. fputs does not allow you to handle this at all, and I checked Bionic, where it does seem that it can do partial writes (at least theoretically), but it will not tell you how many bytes it successfully output - so no way to cleanly recover. It would seem safer to implement FilePuts in terms of fwrite, which does tell you this and allows you to handle it.

Now that I'm having a second look - can we simplify the interface here? I don't see why we are exposing element count * element size to the caller here, as I don't think the calling code really makes an educated choice in their usage. Solving partial reads and writes is easier if we can manage these in the lowest level code. I also don't understand why we're returning the written/read size, as none of the calling code actually do anything useful with it, and we could just as well just be returning a bool.

Reply:
The modifications you are suggesting sound good to me.  Let me know when you have them.  That is weird, BTW, about how that function returns EOF, but thanks for verifying.